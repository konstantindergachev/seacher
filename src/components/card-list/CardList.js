import React from 'react';
import Card from '../card/Card';
import './CardList.scss';

const CardList = ({ cards }) => {
  return (
    <div className="cards">
      {cards.length > 0 ? (
        cards.map((monster) => (
          <Card
            key={monster.id}
            name={monster.name}
            img={monster.img}
            email={monster.email}
          />
        ))
      ) : (
        <span>Loading...</span>
      )}
    </div>
  );
};

export default CardList;
