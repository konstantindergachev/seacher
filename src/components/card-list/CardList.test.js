import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import CardList from './CardList';

describe('CardList component', () => {
  const mockCards = [
    {
      id: '1',
      name: 'John Snow',
      username: 'JohnJohn',
      email: 'john@john.com',
    },
    {
      id: '2',
      name: 'Tom Magnum',
      username: 'Tomas',
      email: 'tom@tom.com',
    },
  ];

  describe('CardList with Card', () => {
    const cardList = shallow(<CardList cards={mockCards} />);
    it('to render CardList component with Card', () => {
      expect(cardList.find('Card')).toHaveLength(2);
    });
    it('to get the snapshot of CardList component', () => {
      const tree = renderer.create(<CardList cards={mockCards} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('CardList with Loader', () => {
    const cardList = shallow(<CardList cards={[]} />);
    it('to render CardList component with Loading', () => {
      expect(cardList.find('span')).toHaveLength(1);
    });
    it('to get the snapshot of CardList component', () => {
      const tree = renderer.create(<CardList cards={[]} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
