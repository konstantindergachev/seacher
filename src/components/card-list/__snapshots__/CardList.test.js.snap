// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList component CardList with Card to get the snapshot of CardList component 1`] = `
<div
  className="cards"
>
  <div
    className="card__container"
  >
    <img
      alt="John Snow"
      className="card__img"
    />
    <div>
      <p
        className="card__name"
      >
        John Snow
      </p>
      <p
        className="card__email"
      >
        john@john.com
      </p>
    </div>
  </div>
  <div
    className="card__container"
  >
    <img
      alt="Tom Magnum"
      className="card__img"
    />
    <div>
      <p
        className="card__name"
      >
        Tom Magnum
      </p>
      <p
        className="card__email"
      >
        tom@tom.com
      </p>
    </div>
  </div>
</div>
`;

exports[`CardList component CardList with Loader to get the snapshot of CardList component 1`] = `
<div
  className="cards"
>
  <span>
    Loading...
  </span>
</div>
`;
